frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -2635,23 +2640,23 @@ static void considerCompilerGeneratedCandidates(Context* context,
   candidates.push_back(instantiated);
 }
 
-static std::vector<BorrowedIdsWithName> lookupCalledExpr(
-    Context* context, const Scope* scope, const CallInfo& ci,
-    NamedScopeSet& visited) {
+static std::vector<BorrowedIdsWithName>
+lookupCalledExpr(Context* context,
```

Comment:
It does not necessarily need to be done in this PR, but I'm reasonably confident we should be able to compose `lookupCalledExprConsideringReceiver` and `lookupCalledExpr` into one function. They contain a good amount of duplicate code including calling out to `gatherReceiverAndParentScopesForType` twice with the same arguments, and `lookupCalledExpr` is only ever called by `lookupCalledExprConsideringReceiver`.

This composition is [something I tried to do in this PR previously](https://github.com/chapel-lang/chapel/pull/21544/commits/2d27c43e91ec64917b1b04d3687c31d6c2db3698#diff-4b6862d44766cb18210d047ec169362a1cb7620a8e7bdc9f4dcf15a1eaaef598R2638) but then [abandoned](https://github.com/chapel-lang/chapel/pull/21544/commits/b8a66cfb9de5e97828d7e0a362a24e0d84d74c0d) as I wanted to cut down my potential surface area for mistakes when debugging.

Reply:
Yes, I agree about that. For now I will just add a TODO comment.

Reply:
To close the loop on this for myself, this is done in https://github.com/chapel-lang/chapel/pull/21555.